migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/migration_impl.go

```diff
@@ -0,0 +1,146 @@
+package m197tom198
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/store/authproviders"
+	"github.com/stackrox/rox/migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/store/groups"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/auth/authproviders/oidc"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+var (
+	log       = logging.LoggerForModule()
+	batchSize = 500
+)
+
+const (
+	orgIDKey            = "orgid"
+	accountIDClaim      = "account_id"
+	groupsKey           = "groups"
+	isOrgAdminKey       = "is_org_admin"
+	orgAdminGroupsValue = "org_admin"
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	groupStore := groups.New(database.PostgresDB)
+	authProviderStore := authproviders.New(database.PostgresDB)
+	oidcAuthProviders := make([]*storage.AuthProvider, 0, batchSize)
+
+	err := authProviderStore.Walk(ctx, func(obj *storage.AuthProvider) error {
+		if obj.GetType() == oidc.TypeName {
+			oidcAuthProviders = append(oidcAuthProviders, obj)
+			if len(oidcAuthProviders) == batchSize {
+				if err := migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders); err != nil {
+					return err
+				}
+				oidcAuthProviders = oidcAuthProviders[:0]
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders)
+}
+
+func migrateAuthProviderClaims(ctx context.Context, groupStore groups.Store, providerStore authproviders.Store, providers []*storage.AuthProvider) error {
+	providerToGroups := make(map[string][]*storage.Group, len(providers))
+	for _, provider := range providers {
+		providerToGroups[provider.GetId()] = make([]*storage.Group, 0)
+	}
+	err := groupStore.Walk(ctx, func(obj *storage.Group) error {
+		authProviderID := obj.GetProps().GetAuthProviderId()
+		if _, ok := providerToGroups[authProviderID]; ok {
+			providerToGroups[authProviderID] = append(providerToGroups[authProviderID], obj)
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to get groups from the store")
+	}
+	migratedProviders := make([]*storage.AuthProvider, 0)
```

Comment:
nit: for groups I get the 0 as len, here we can use `len(providers)` again.

Reply:
I think I don't get it. `make([]*storage.AuthProvider, len(providers)` will create an array with `len(providers)` `nil` elements. `make([]*storage.AuthProvider, 0, len(providers)`  will create an array with 0 elements, but `len(providers)` capacity. I would expect the number of `migratedProviders` to be quite low, so there's no need to set capacity to `len(providers` on creation.